rules/manager_test.go

```diff
@@ -2112,6 +2112,139 @@ func TestAsyncRuleEvaluation(t *testing.T) {
 	})
 }
 
+func TestNewRuleGroupRestoration(t *testing.T) {
```

Comment:
Thanks for the tests, nice addition.
I ran them on my machine and they took `2-4s` each, if you could make them runnable in parallel it'd be great.